python/tests/core/constraints/factories/test_distribution_metrics.py

```diff
@@ -1,13 +1,65 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import whylogs as why
+from whylogs.core.constraints import ConstraintsBuilder
 from whylogs.core.constraints.factories.distribution_metrics import (
     greater_than_number,
+    is_in_range,
+    is_non_negative,
     mean_between_range,
     quantile_between_range,
     smaller_than_number,
     stddev_between_range,
 )
 
 
-def test_greater_than_number(builder, nan_builder):
+def test_is_in_range(builder, nan_builder):
+    builder.add_constraint(is_in_range(column_name="weight", lower=1.1, upper=3.2, skip_missing=False))
+    builder.add_constraint(is_in_range(column_name="legs", lower=0, upper=6, skip_missing=False))
+    builder.add_constraint(is_in_range(column_name="animal", lower=0, upper=6, skip_missing=False))
+
+    constraint = builder.build()
+    report = constraint.generate_constraints_report()
+    assert report[0].name == "weight is in range [1.1,3.2]" and report[0].passed == 0
+    assert report[1].name == "legs is in range [0,6]" and report[1].passed == 1
+    assert report[2].name == "animal is in range [0,6]" and report[2].passed == 0
+
+    nan_builder.add_constraint(is_in_range(column_name="a", lower=1.1, upper=3.2, skip_missing=True))
+    constraint = nan_builder.build()
+    report = constraint.generate_constraints_report()
+    assert report[0].name == "a is in range [1.1,3.2]" and report[0].passed == 1
+
+
+def test_is_non_negative():
+    data = {
+        "animal": ["cat", "hawk", "snake", "cat", "mosquito"],
+        "legs": [4, 2, 0, 4, 6],
+        "weight": [4.3, -1.8, 1.3, 4.1, 5.5e-6],
+        "null_column": [np.nan, np.nan, np.nan, np.nan, np.nan],
+    }
+
+    view = why.log(pd.DataFrame(data)).profile().view()
+    builder = ConstraintsBuilder(dataset_profile_view=view)
+
+    builder.add_constraint(is_non_negative(column_name="weight"))
+    builder.add_constraint(is_non_negative(column_name="legs"))
+    builder.add_constraint(is_non_negative(column_name="animal", skip_missing=False))
+    builder.add_constraint(is_non_negative(column_name="null_column", skip_missing=True))
+    with pytest.raises(ValueError) as e:
```

Comment:
```suggestion
    with pytest.raises(ValueError, match="non_existing_column was not found in set of this profile") as e:
```

Then you can drop the assert on line 52

Reply:
done! thanks!